netket/sampler/metropolis.py

```diff
@@ -375,8 +375,7 @@ def _init_state(sampler, machine, params, key):
         # now.
         if not sampler.reset_chains:
             key_state, rng = jax.random.split(key_state)
-            with jax.spmd_mode("allow_all"):
-                σ = sampler.rule.random_state(sampler, machine, params, state, rng)
+            σ = sampler.rule.random_state(sampler, machine, params, state, rng)
```

Comment:
No, I'm not sure you can do that. There are potentially other rules that are not jitted (we do not guarantee that random rule is fitted. Just that it's jax-jittable). For example if someone defines their own Hilbert space.

I think we should jit the full `reset` function, which would address all those issues,.